ui/dashboards/src/components/Variables/TemplateVariable.tsx

```diff
@@ -16,25 +16,27 @@ import { Select, FormControl, InputLabel, MenuItem, Box, LinearProgress, TextFie
 import { DEFAULT_ALL_VALUE, ListVariableDefinition, VariableName, VariableValue } from '@perses-dev/core';
 import { useTemplateVariable, useTemplateVariableActions } from '../../context';
 import { useListVariablePluginValues } from './variable-model';
+
 type TemplateVariableProps = {
   name: VariableName;
+  source?: string;
```

Comment:
I like that this is optional but could it be more descriptive than `string`?

From the tests and discussion #226, it sounds like `'project' | 'global'` are the two expected values. Possible we could add 'dashboard' too and make that the default but I guess leaving undefined is better so `source` doesn't get set in the JSON unnecessarily.

Reply:
So ignore my 'dashboard' idea, maybe this could work and TemplateVariableSource could be exported out of core package since looks like it's used in a few other files

```typescript
export type TemplateVariableSource = 'project' | 'global';

type TemplateVariableProps = {
  name: VariableName;
  source?: TemplateVariableSource;
};
```

Reply:
Yeah true, that’s a good call!

if/when we implement the spec at Chronosphere, we’d likely want to change ‘project’ to ‘collection’ since that’s our app’s info architecture